### DISCUSSION TITLE
Create a custom panel for 3d visualizations

### DISCUSSION THREAD

**Discussion by ghost** [foxglove/community#782] [Q&A]
Hello,
I want to create a custom panel which could visualize the foxglove.SceneUpdate schema. Is this possible?
I know the built-in 3D panel can visualize this topic, however, I want to have some user interaction on selecting an object via the use of `selected_id`. This is not possible in a simple message converter extension using built-in 3D panel.

I am wondering if this is possible at all? The example of a simple-3D-extension does not help much.
or is something like this possible https://github.com/orgs/foxglove/discussions/713 but not for publishing points but rather publishing `selected_id`?

**Comment by defunctzombie** [comment]
Yes - it is possible to create a custom panel. Check out our guides here: https://foxglove.dev/docs/studio/extensions/getting-started#custom-panels
